### PR TITLE
[235] add qa aks env

### PR DESCRIPTION
**Pull request by smneal**
### Context
Add QA environment to the AKS platform

### Changes proposed in this pull request

- [Terraform changes to support additional envs](https://github.com/DFE-Digital/publish-teacher-training/pull/3644/commits/e8d767ad72874e0ccbb02d5878f026f1d7ace2a9)
- [Add qa_aks Terraform config](https://github.com/DFE-Digital/publish-teacher-training/pull/3644/commits/ea7cf0ae5a745a0921d4c4b49b824bf61b905fee)
- [Add Ruby config to support qa_aks](https://github.com/DFE-Digital/publish-teacher-training/pull/3644/commits/0c2fdfcd276d3e02a45324741db6ab68d425ab6d)
- [Add Makefile and config to support qa_aks](https://github.com/DFE-Digital/publish-teacher-training/pull/3644/commits/ecd66b3e0c340ed5c38ea1137d804f6313622850)
- [Update GitHub actions to support qa_aks](https://github.com/DFE-Digital/publish-teacher-training/pull/3644/commits/b6e17d1a129ae3cdb5362efc190b2aa7116786ea)
- [Add AKS database restore workflow](https://github.com/DFE-Digital/publish-teacher-training/pull/3644/commits/f082d18e5b1ea435177a29297b2536aa99f4b7e5)
- [DNS changes to support qa_aks](https://github.com/DFE-Digital/publish-teacher-training/pull/3644/commits/06c9b5a1b45df48e7fa563b0385f5b80892569f9)

### Guidance to review

- [ ] [Deployment to qa_aks complete](https://github.com/DFE-Digital/publish-teacher-training/actions/runs/5254525111/jobs/9493357334)
- [ ] [Restore AKS database complete](https://github.com/DFE-Digital/publish-teacher-training/actions/runs/5253466759)
- [ ] Find accessible at: https://qa2.find-postgraduate-teacher-training.service.gov.uk
- [ ] Publish accessible at: https://qa2.publish-teacher-training-courses.service.gov.uk
- [ ] Publish API accessible at: https://qa2.publish-teacher-training-courses.service.gov.uk/docs/#about
- [ ] Validation Trello [card](https://trello.com/c/0IcaV60o) passed

### Checklist

- [ ] Make sure all information from the Trello card is in here
- [ ] Attach to Trello card
- [ ] Rebased main
- [ ] Cleaned commit history
- [ ] Tested by running locally
- [ ] Inform data insights team due to database changes

### CODE REVIEW

**Comment by smneal** [comment]
[FindPub - Replicate each environment to AKS](https://trello.com/c/NXkjlzI0/235-findpub-replicate-each-environment-to-aks)

File: config/settings/qa_aks.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+gcp_api_key: please_change_me
+publish_api_url: https://qa2.api.publish-teacher-training-courses.service.gov.uk
+publish_url: https://qa2.publish-teacher-training-courses.service.gov.uk
+find_url: https://qa2.find-postgraduate-teacher-training.service.gov.uk
+find_assets_url: https://qa2.find-postgraduate-teacher-training.service.gov.uk
```
Comment (RMcVelia) [inline config/settings/qa_aks.yml:5]: would we just drop find_assets_url as caching would be done via the fd route?

File: terraform/aks/workspace_variables/qa_aks.tfvars.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  "enable_find": true,
+  "additional_hostnames": [
+    "qa2.publish-teacher-training-courses.service.gov.uk",
+    "qa2.find-postgraduate-teacher-training.service.gov.uk",
+    "qa2-assets.find-postgraduate-teacher-training.service.gov.uk",
+    "qa2.api.publish-teacher-training-courses.service.gov.uk",
+    "find-qa.test.teacherservices.cloud"
+  ],
+  "enable_monitoring": false
+}
```
Comment (RMcVelia) [inline terraform/aks/workspace_variables/qa_aks.tfvars.json:30]: azure_patch_schedule for redis (although can go in the .tf file

File: terraform/aks/workspace_variables/qa_aks.tfvars.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  "worker_apps": {
+    "worker": {
+      "replicas": 1
+    }
+  },
+  "enable_find": true,
+  "additional_hostnames": [
+    "qa2.publish-teacher-training-courses.service.gov.uk",
+    "qa2.find-postgraduate-teacher-training.service.gov.uk",
+    "qa2-assets.find-postgraduate-teacher-training.service.gov.uk",
```
[the remark this answers is not in the document]
Comment (smneal) [reply terraform/aks/workspace_variables/qa_aks.tfvars.json:25]: See previous :)

**Review by defong — COMMENTED** [review]
This looks fine to me, deferring to @avinhurry @JR-G for so they are aware

File: terraform/aks/workspace_variables/qa_aks.tfvars.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (smneal) [reply terraform/aks/workspace_variables/qa_aks.tfvars.json:25]: I've dropped this now and `find-qa` which is also not actually needed either

File: terraform/aks/workspace_variables/qa_aks.tfvars.json (hunk 2 above)
Comment (smneal) [reply terraform/aks/workspace_variables/qa_aks.tfvars.json:28]: Added to tf file

**Review by RMcVelia — APPROVED** [review]

**Review by JR-G — APPROVED** [review]

**Pull request merged**